### PR TITLE
Issue #6207: Added Xpath IT regression test for MagicNumber

### PR DESCRIPTION
**Pull request by Rohanraj123**
Part of #6207 

Documentation : [MagicNumber](https://checkstyle.org/checks/coding/magicnumber.html)

### CODE REVIEW

**Comment by nrmancuso** [comment]
@Rohanraj123 Please read https://github.com/checkstyle/checkstyle/issues/6207#issue-379425252 and do all that it requires.

**Comment by Rohanraj123** [comment]
> @Rohanraj123 Please read [#6207 (comment)](https://github.com/checkstyle/checkstyle/issues/6207#issue-379425252) and do all that it requires.

So as per my understanding yet , I only need to add a link of the documentation . Right? Else I think I have dont correctly @nrmancuso  Sir.

**Comment by romani** [comment]
>XPath test must have 3 tests

Please add one more test

**Comment by nrmancuso** [comment]
Check has one message: https://checkstyle.org/checks/coding/magicnumber.html#Violation_Messages

We should have one test.

**Comment by Rohanraj123** [comment]
> > XPath test must have 3 tests
> 
> Please add one more test

@romani Sir, Although its not necessarily required. It while mvn clean verify it says that two or more testcases are required for unit testing in Xpath. And i feel two tests are enough to test for magic number. But still if I can add one more test case. Please let me know.

**Comment by romani** [comment]
@Rohanraj123 , thanks a lot for raising this unclear point, please do 3. I updated issue description.

**Comment by Rohanraj123** [comment]
> @Rohanraj123 , thanks a lot for raising this unclear point, please do 3. I updated issue description.

Sure !

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.checkstyle.suppressionxpathfilter.magicnumber;
+
+public class SuppressionXpathRegressionMagicNumberAnotherVariable {
+    int a = 20; // warn
```
[the remark this answers is not in the document]
Comment (Rohanraj123) [reply src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java]: Added a test file with the pointers given by you @romani Sir.

File: src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package org.checkstyle.suppressionxpathfilter.magicnumber;
+
+public class SuppressionXpathRegressionMagicNumberAnotherVariable {
+    public void performOperation() {
+        try {
+            int result = performCalculation();
+        } catch(Exception e) {
+            handleException(e);
```
Comment (romani) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java:8]: make it: 
```
} catch(Exception e) {
        int a = 0;
        boolean someCondition = false;
        if (someCondition) {
            a = 1; // ok
        } else {
            a = 20; // warn
        }
```
to let generated xpath be long and complicated

File: src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        }
+    }
+    private int performCalculation() throws Exception {
```
Comment (romani) [inline src/it/resources/org/checkstyle/suppressionxpathfilter/magicnumber/SuppressionXpathRegressionMagicNumberAnotherVariable.java:11]: remove this method it does not contribute to generated xpath

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Review by romani — APPROVED** [review]
Ok to merge if ci pass

**Pull request merged**